app/com/arpnetworking/metrics/portal/scheduling/impl/JobScheduler.java

```diff
@@ -0,0 +1,225 @@
+                        // It's not time to execute the job yet, but it might be soon.
+                        // Let's ensure we wake up "exactly" when it's time to execute the job,
+                        //   else we might be up to 1tick late executing it.
+                        //   (Well-- "exactly" is a tall order, but we can at least ensure we wake up
+                        //    _very slightly_ late.)
+                        timeUntilExtraTick(_clock.instant(), nextRun).ifPresent(delta -> {
+                            timers().startSingleTimer("TICK_ONEOFF", Tick.INSTANCE, delta);
+                        });
+                    } else {
+                        final ActorRef sender = getSender();
+                        PatternsCS.pipe(
+                                job.execute(getSelf(), nextRun)
+                                        .handle((result, error) -> {
+                                            if (error == null) {
+                                                _jobRef.jobSucceeded(nextRun, result);
+                                                return new Success<>(_jobRef, nextRun, result);
+                                            } else {
+                                                _jobRef.jobFailed(nextRun, error);
+                                                return new Failure<>(_jobRef, nextRun, error);
+                                            }
+                                        }),
+                                getContext().dispatcher())
+                                .to(sender);
+                        _jobRef.jobStarted(nextRun);
+                    }
+                })
+                .build();
+    }
+
+    protected static final FiniteDuration TICK_INTERVAL = Duration.apply(1, TimeUnit.MINUTES);
+    protected static final FiniteDuration SLEEP_SLOP = Duration.apply(10, TimeUnit.MILLISECONDS);
+    private static final Logger LOGGER = LoggerFactory.getLogger(JobScheduler.class);
+
+    /**
+     * Internal message, telling the scheduler to run any necessary jobs.
+     */
+    protected static final class Tick {
+        public static final Tick INSTANCE = new Tick();
+    }
+
+    /**
+     *
+     * @param <T>
+     */
+    public static final class Success<T> implements Serializable {
+        private final JobRef<T> _jobRef;
+        private final Instant _scheduled;
+        private final T _result;
+
+        Success(final JobRef<T> jobRef, final Instant scheduled, final T result) {
+            _jobRef = jobRef;
+            _scheduled = scheduled;
+            _result = result;
+        }
+
+        public JobRef<T> getJobRef() {
+            return _jobRef;
+        }
+
+        public Instant getScheduled() {
+            return _scheduled;
+        }
+
+        public T getResult() {
+            return _result;
+        }
+
+        private static final long serialVersionUID = 1L;
+    }
+
+    /**
+     * @param <T>
```

Comment:
Incomplete; although again, not sure this needs to be public.

Reply:
I guess I can make it `protected` until a reason for publicity presents itself. It seems useful to let the executor notify people when it completes (e.g. for tests with `TestKit`).